cpp/tests/binaryop/assert-binops.h

```diff
@@ -37,15 +37,15 @@ void ASSERT_BINOP(column_view const& out,
                   scalar const& lhs,
                   column_view const& rhs,
                   TypeOp&& op) {
-    auto lhs_h = static_cast<const ScalarType*>(&lhs)->value();
+    auto lhs_h = static_cast<ScalarType const&>(lhs).operator TypeLhs();
```

Comment:
Ah, so this was tricky. Most of the time `TypeLhs` is the same between host and device (int32_t etc.).

But for strings, we need `std::string` on the host, and `cudf::string_view` on device.

So most of the time this calls the new `scalar value_type()` conversion operator, but for strings it calls the `string_scalar std::string()` conversion operator.

This disparity is because the `cudf::string_view` is a non-owning type, so we can't use it as the type parameter in the `to_host` calls here.

This means `TypeLhs`, `TypeRhs`, and `TypeOut` are really the _host_ types, not device types, even though most of the time the host and device types are the same.